src/dc_receive_imf.rs

```diff
@@ -227,6 +229,19 @@ pub async fn dc_receive_imf(
             context
                 .do_heuristics_moves(server_folder.as_ref(), insert_msg_id)
                 .await;
+            if !mime_parser.mdn_reports.is_empty() && is_dc_message != MessengerMessage::No {
```

Comment:
Maybe check for `mime_parser.has_chat_version()` here instead of `is_dc_message != MessengerMessage::No` and avoid passing `is_dc_message` around? This will save us output `&mut` argument and simplify the logic.

Reply:
Makes sense.